packages/base/src/panelview/formbuilder.tsx

```diff
@@ -202,8 +219,92 @@ export class ObjectPropertiesForm extends React.Component<IProps, IStates> {
           </div>
         </div>
       );
-    } else {
-      return <div>{this.buildForm()}</div>;
     }
   }
 }
+
+export class LayerPropertiesForm extends ObjectPropertiesForm {
+  protected processSourceData(sourceData: IDict<any>) {
+    // Replace the source id by its name in the form
+    sourceData.source = this.props.model.getSource(sourceData.source)?.name;
+  }
+
+  protected syncData(properties: IDict<any>): void {
+    // Replace selected source name by its id
+    const sources = this.props.model.getSources();
+    if (!sources) {
+      throw Error('Unreachable');
+    }
+
+    for (const source of Object.keys(sources)) {
+      if (sources[source].name === properties.source) {
+        properties.source = source;
+        break;
+      }
+    }
+
+    super.syncData(properties);
+  }
+
+  protected processSchema(
+    data: IDict<any> | undefined,
+    schema: IDict<any>
+  ): void {
+    super.processSchema(data, schema);
+
+    // Replace the source text box by a dropdown menu
+    const sourceNames: string[] = [];
+    for (const sourceId of Object.keys(this.props.model.getSources() || {})) {
+      const source = this.props.model.getSource(sourceId);
+      if (source) {
+        sourceNames.push(source.name);
+      }
+    }
+
+    if (schema.properties.source) {
+      schema.properties.source.enum = sourceNames;
```

Comment:
The enum can have `enumName` (display) and `enum` (value(s)), in this case respectively the source name and source id.

Reply:
Right thanks, I'll update this